mlflow/transformers.py

```diff
@@ -0,0 +1,990 @@
+            "transformers model.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+
+@experimental
+def get_default_conda_env(model):
+    """
+    :return: The default Conda environment for MLflow Models produced with the ``transformers``
+             flavor, based on the model instance framework type of the model to be logged.
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements(model))
+
+
+@experimental
+@docstring_version_compatibility_warning(integration_name=FLAVOR_NAME)
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
+def save_model(
+    transformers_model,
+    path: str,
+    processor=None,
+    task: Optional[str] = None,
+    model_card=None,
+    code_paths: Optional[List[str]] = None,
+    mlflow_model: Optional[Model] = None,
+    signature: Optional[ModelSignature] = None,
+    input_example: Optional[ModelInputExample] = None,
+    pip_requirements: Optional[Union[List[str], str]] = None,
+    extra_pip_requirements: Optional[Union[List[str], str]] = None,
+    conda_env=None,
+    metadata: Dict[str, Any] = None,
+    **kwargs,
+) -> None:
+    """
+    Save a trained transformers model to a path on the local file system.
+
+    :param transformers_model:
+        A trained transformers `Pipeline` or a dictionary that maps required components of a
+        pipeline to the named keys of ["model", "image_processor", "tokenizer",
+        "feature_extractor"]. The `model` key in the dictionary must map to a value that inherits
+        from `PreTrainedModel`, `TFPreTrainedModel`, or `FlaxPreTrainedModel`.
+        All other component entries in the dictionary must support the defined task type that is
+        associated with the base model type configuration.
+
+        An example of supplying component-level parts of a transformers model is shown below:
+
+        .. code-block:: python
+
+          from transformers import MobileBertForQuestionAnswering, AutoTokenizer
+
+          architecture = "csarron/mobilebert-uncased-squad-v2"
+          tokenizer = AutoTokenizer.from_pretrained(architecture)
+          model = MobileBertForQuestionAnswering.from_pretrained(architecture)
+
+          with mlflow.start_run():
+              components = {
+                  "model": model,
+                  "tokenizer": tokenizer,
+              }
+              mlflow.transformers.save_model(
+                  transformers_model=components,
+                  path="path/to/save/model",
+              )
+
+        An example of submitting a `Pipeline` from a default pipeline instantiation:
+
+        .. code-block:: python
+
+          from transformers import pipeline
+
+          pipeline = "csarron/mobilebert-uncased-squad-v2"
+
+          with mlflow.start_run():
+              mlflow.transformers.save_model(
+                  transformers_model=pipeline,
+                  path="path/to/save/model",
+              )
```

Comment:
This seems to fail:

```
Traceback (most recent call last):
  File "save.py", line 6, in <module>
    mlflow.transformers.save_model(
  File "/Users/corey.zumar/mlflow/mlflow/utils/docstring_utils.py", line 235, in version_func
    return func(*args, **kwargs)
  File "/Users/corey.zumar/mlflow/mlflow/transformers.py", line 275, in save_model
    _validate_transformers_model_dict(transformers_model)
  File "/Users/corey.zumar/mlflow/mlflow/transformers.py", line 141, in _validate_transformers_model_dict
    model = transformers_model.model
AttributeError: 'str' object has no attribute 'model'
```

Reply:
(Transformers version 4.27.3)

Reply:
I.... don't even know... what I was thinking with that.... UGH. Fix incoming.